### PR TITLE
Delete ngraph `AlignedBuffer`/`SharedBuffer`

### PR DESCRIPTION
**Pull request by vurusovs**

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]
please, fix clang format issues

File: src/core/src/runtime/aligned_buffer.cpp
Comment (ilya-lavrenov) [inline src/core/src/runtime/aligned_buffer.cpp:1]: let's keep this file

Comment (vurusovs) [reply src/core/src/runtime/aligned_buffer.cpp:1]: Done

File: src/core/include/openvino/op/constant.hpp [hunk 1]
```diff
@@ -12,7 +12,8 @@
 #    define WAS_OV_LIBRARY_DEFINED_CONSTANT
 #endif
 
-#include "ngraph/runtime/shared_buffer.hpp"
+#include "ngraph/util.hpp"
```
Comment (praasz) [inline src/core/include/openvino/op/constant.hpp:15]: Can be removed?

File: src/core/src/runtime/aligned_buffer.cpp [hunk 2]
```diff
@@ -2,70 +2,12 @@
...
-}
-}  // namespace ngraph
-
-namespace ov {
-AttributeAdapter<std::shared_ptr<ngraph::runtime::AlignedBuffer>>::AttributeAdapter(
-    std::shared_ptr<ngraph::runtime::AlignedBuffer>& value)
-    : DirectValueAccessor<std::shared_ptr<ngraph::runtime::AlignedBuffer>>(value) {}
-}  // namespace ov
-NGRAPH_SUPPRESS_DEPRECATED_END
+// #include "openvino/util/log.hpp"
```
Comment (praasz) [inline src/core/src/runtime/aligned_buffer.cpp:10]: Remove ?

File: src/plugins/intel_cpu/tests/functional/subgraph_tests/src/denormal_check.cpp [hunk 3]
```diff
@@ -6,36 +6,14 @@
...
-        return _buffer->size() / sizeof(T);
-    }
-private:
-    std::unique_ptr<ngraph::runtime::AlignedBuffer> _buffer = nullptr;
-};
-
 class DenormalNullifyCheck : public SubgraphBaseTest {
 protected:
-std::unique_ptr<AlignedBufferWrapper<float>> pConstStorage;
+std::shared_ptr<ov::AlignedBuffer> pConstStorage;
```
Comment (praasz) [inline src/plugins/intel_cpu/tests/functional/subgraph_tests/src/denormal_check.cpp:16]: Why use shared not unique?

File: src/core/include/openvino/op/constant.hpp (hunk 1 above)
Comment (vurusovs) [reply src/core/include/openvino/op/constant.hpp:15]: I did it in https://github.com/openvinotoolkit/openvino/pull/22184, but the change requires a lot of fixes

File: src/plugins/intel_cpu/tests/functional/subgraph_tests/src/denormal_check.cpp (hunk 3 above)
Comment (vurusovs) [reply src/plugins/intel_cpu/tests/functional/subgraph_tests/src/denormal_check.cpp]: Reverted to unique_ptr

File: src/core/src/runtime/aligned_buffer.cpp [hunk 4]
```diff
@@ -2,70 +2,12 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include "ngraph/runtime/aligned_buffer.hpp"
-
 #include <algorithm>
 #include <memory>
 
-#include "ngraph/util.hpp"
+// #include "ngraph/util.hpp"
```
[the remark this answers is not in the document]
Comment (vurusovs) [reply src/core/src/runtime/aligned_buffer.cpp]: Done, thanks!

File: src/core/src/runtime/aligned_buffer.cpp (hunk 2 above)
Comment (vurusovs) [reply src/core/src/runtime/aligned_buffer.cpp]: Done, thanks!

**Pull request merged**